lib/puppet/type/wait_for.rb

```diff
@@ -78,14 +94,25 @@ def sync
           end
         end
       rescue Timeout::Error
-        self.fail Puppet::Error, "Query exceeded timeout", $!
+        self.fail Puppet::Error, "Exceeded timeout", $!
       end
 
       unless status
         if self.class == Puppet::Type::Wait_for::Exit_code
           self.fail Puppet::Error, "Exit status #{@output.exitstatus.to_i} after max_retries"
         elsif self.class == Puppet::Type::Wait_for::Regex
           self.fail Puppet::Error, "Did not match regex after max_retries"
+        elsif self.class == Puppet::Type::Wait_for::State
```

Comment:
In so far as you appear to be able to test the value of `self.class == Puppet::Type::Wait_for::State` will it not be more consistent and also simpler if you test that in the block above and avoid the need for the new if statement?

Reply:
Unfortunately not, because of the need to execute the command when in "query" mode.